pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -500,7 +500,28 @@ def set_distributed_mode(self, distributed_backend: Optional[str] = None):
             else:
                 rank_zero_warn('You are running on single node with no parallelization, so distributed has no effect.')
                 self._distrib_type = None
-
+        
+        # finished configuring self._distrib_type, check ipython environment
+        ipython_incompatible_distrib_types = [DistributedType.DDP, DistributedType.DDP2, DistributedType.HOROVOD]
+        if self._distrib_type in ipython_incompatible_distrib_types:
+            # check ipython env
```

Comment:
Could you put this logic into its own function ? Like `_check_ipython_compatible`

Reply:
Done, not sure the naming fits the convention, but I see `check_horovod()` is okay.